docs/sql-ref-syntax-aux-describe-table.md

```diff
@@ -60,8 +61,95 @@ to return the metadata pertaining to a partition or column respectively.
+  
+  Below are the schema definitions for `<type_json>`:
+
+| Spark SQL Data Types  | JSON Representation                                                                                                   |
+|-----------------------|-----------------------------------------------------------------------------------------------------------------------|
+| ByteType              | `{ "type" : "tinyint" }`                                                                                              |
+| ShortType             | `{ "type" : "smallint" }`                                                                                             |
+| IntegerType           | `{ "type" : "int" }`                                                                                                  |
+| LongType              | `{ "type" : "bigint" }`                                                                                               |
+| FloatType             | `{ "type" : "float" }`                                                                                                |
+| DoubleType            | `{ "type" : "double" }`                                                                                               |
+| DecimalType           | `{ "type": "decimal", "precision": p, "scale": s }`                                                                   |
+| StringType            | `{ "type" : "string" }`                                                                                               |
+| VarCharType           | `{ "type" : "varchar", "length": n }`                                                                                 |
+| CharType              | `{ "type" : "char", "length": n }`                                                                                    |
+| BinaryType            | `{ "type" : "binary" }`                                                                                               |
+| BooleanType           | `{ "type" : "boolean" }`                                                                                              |
+| DateType              | `{ "type" : "date" }`                                                                                                 |
+| TimestampType         | `{ "type" : "timestamp_ltz" }`                                                                                        |
+| TimestampNTZType      | `{ "type" : "timestamp_ntz" }`                                                                                        |
+| YearMonthIntervalType | `{ "type" : "interval", "start_unit": "<start_unit>", "end_unit": "<end_unit>" }`                                     |
+| DayTimeIntervalType   | `{ "type" : "interval", "start_unit": "<start_unit>", "end_unit": "<end_unit>" }`                                     |
+| ArrayType             | `{ "type" : "array", "element_type": <type_json>, "contains_null": <boolean> }`                                       |
```

Comment:
contains_null doesn't sound right. There is a difference between actually containing a NULL (which is information about actual data) and a nulls being allowed or not. We could choose nullable, not_nullable, or not_null for example.

Reply:
`contains_null` comes from the ArrayType field naming in the [docs page](https://spark.apache.org/docs/4.0.0-preview2/sql-ref-datatypes.html)

> ArrayType(elementType, [containsNull])